UI/WebServerResources/SOGoMailDataSource.js

```diff
@@ -82,9 +82,25 @@ SOGoMailDataSource = Class.create({
         },
 
         _loadCallback: function(http) {
+            var data, re;
             if (http.status == 200) {
                 if (http.responseText.length > 0) {
-                    var data = http.responseText.evalJSON(true);
+                    try {
+                        data = http.responseText.evalJSON(true);
+                    } catch(e) {
+                        if (e instanceof SyntaxError) {
+                            re = /action=["']\/SOGo\/connect["']/i;
+                            if (re.test(http.responseText)) {
+                                window.location= '/SOGo';
```

Comment:
I will look then for other string in the login form. However the solution will always be inherently fragile, the good one would be to change SOGo so in this AJAX call he return a error code instead of the login form. But we will not do this.